Sources/SnapshotTesting/Snapshotting/UIImage.swift

```diff
@@ -4,16 +4,24 @@ import XCTest
 
 extension Diffing where Value == UIImage {
   /// A pixel-diffing strategy for UIImage's which requires a 100% match.
-  public static let image = Diffing.image(precision: 1)
+  public static let image = Diffing.image(precision: 1, scale: nil)
 
   /// A pixel-diffing strategy for UIImage that allows customizing how precise the matching must be.
   ///
   /// - Parameter precision: A value between 0 and 1, where 1 means the images must match 100% of their pixels.
+  /// - Parameter scale: Scale to use when loading the reference image from disk. If `nil` or the `UITraitCollection`s default value of `0.0`, the screens scale is used.
   /// - Returns: A new diffing strategy.
-  public static func image(precision: Float) -> Diffing {
+  public static func image(precision: Float, scale: CGFloat?) -> Diffing {
+    let imageScale: CGFloat
+    if let scale = scale, scale != 0.0 {
+      imageScale = scale
+    } else {
+        imageScale = UIScreen.main.scale
+    }
+
     return Diffing(
       toData: { $0.pngData() ?? emptyImage().pngData()! },
-      fromData: { UIImage(data: $0, scale: UIScreen.main.scale)! }
+      fromData: { UIImage(data: $0, scale: imageScale)! }
```

Comment:
Looks like the default undefined value is now `nil` instead of `0`, right? In that case could you remove the logic above and instead change this single line to:

```swift
      fromData: { UIImage(data: $0, scale: scale ?? UIScreen.main.scale)! }
```

Reply:
I introduced this logic to handle the case of a `UITraitCollection` without a `displayScale` set being passed in. With such a trait collection, the default value of the `displayScale` is `0.0` ([see here](https://developer.apple.com/documentation/uikit/uitraitcollection/1623519-displayscale)).

Let me know if I didn't understand your concern right 🙂

Reply:
Ah yup that makes sense! Thanks!